### ISSUE TITLE
Create rich authorization hooks for Flight gRPC service

### ISSUE THREAD

**Issue by nbauernfeind** [deephaven/deephaven-core#3146]
First pass I implemented flight as a grpc service, but that appears to be short sighted.
1) the messages are not protobufs; they are custom marshalled BarrageMessages (in some cases)
2) many operations operate on specific tickets that should be resolved before authorization (for consistency with table services)
3) some methods are bidirectional (subscriptions start before we know what table is being subscribed to)

At the moment, data that is exposed via flight can only be controlled via the auth transformation.

**Comment by rcaudy** [comment]
Can you elaborate on the proposed change? Transformations seem rich enough for the use cases I'm aware of.

**Comment by nbauernfeind** [comment]
This would enable/disable subscriptions/features based on the Table instances. Devin mentioned he may want to disable DoPut.

**Comment by devinrsmith** [comment]
Yes - I'm not sure the semantics of DoExchange - but I'd also like to be able to limit "pushing" from client -> server via DoExchange.
